### PR TITLE
[Core][Geometry] New IsInsideLocalSpace function

### PR DESCRIPTION
**Pull request by tteschemacher**
Hi all,

this is the first interface coming from #5781 .

It provides the split up of the old IsInside and extracts a new one IsInsideLocalCoordinates which can be optimized for standard cases. The derived classes are not affected by this interface.

Thx!

### CODE REVIEW

File: kratos/geometries/geometry.h [hunk 1]
```diff
@@ -1372,21 +1372,49 @@ class Geometry
...
+    /**
+    * @brief Checks if given point in local space coordinates of this geometry
+    *        is inside the geometry boundaries.
+    * @param rPointLocalCoordinates the point on the geometry,
+    *        which shall be checked if it lays within
+    *        the boundaries.
+    * @param Tolerance the tolerance to the boundary.
+    * @return true if the point is inside, false otherwise
+    */
+    virtual bool IsInsideLocalSpace(
```
Comment (loumalouomega) [inline kratos/geometries/geometry.h]: What the difference with IsInside?

Comment (loumalouomega) [reply kratos/geometries/geometry.h]: The local coordinates?

Comment (tteschemacher) [reply kratos/geometries/geometry.h]: IsInside requires global coordinates. For some cases this projection needs to be done outside in advance or whatever. Thus, for some scenarios using this function is more optimized.

You can check the IsInside functionhow it is defined right now, then you can see the difference.

**Review by loumalouomega — DISMISSED** [review]
OK

File: kratos/geometries/geometry.h (hunk 1 above)
Comment (tteschemacher) [reply kratos/geometries/geometry.h]: @pooyan-dadvand shall we return here already the int?
0 -> outside
1 -> inside
2 -> boundary

Comment (pooyan-dadvand) [reply kratos/geometries/geometry.h]: yep!

**Review by pooyan-dadvand — APPROVED** [review]
For me it is what we discussed in @KratosMultiphysics/technical-committee

**Pull request merged**